packages/core/src/Response.ts

```diff
@@ -1,16 +1,27 @@
 import type { NodeWithScore } from "./Node.js";
+import type { ChatMessage, ChatResponse } from "./llm/types.js";
 
-/**
- * Response is the output of a LLM
- */
-export class Response {
+export class Response implements ChatResponse {
```

Comment:
Idea: we include `message` and `raw` from `ChatResponse` instead of adding `chatResponse: ChatResponse` as an attribute to `Response`

Note: we `ChatResponse` supports extension with `Options` we have to add this to this PR (after we agree on the contract)

Reply:
another advantage is that `Reponse` behaves like `ChatReponse`

Reply:
TBD: how to deal with `chunk`s? We have `ChatReponseChunk`s as type returned by LLMs but chat engines are just streaming `Response` objects. 

Suggestion: we add an optional `delta` attribute to `Response`